projects/Rockchip/devices/RK356X/options

```diff
@@ -0,0 +1,34 @@
+################################################################################
+# setup device defaults
+################################################################################
+
+  # The TARGET_CPU variable controls which processor should be targeted for
+  # generated code.
+    case $TARGET_ARCH in
+      aarch64)
+        TARGET_CPU="cortex-a55"
+        TARGET_CPU_FLAGS="+crc+crypto"
+        ;;
+      arm)
+        TARGET_KERNEL_ARCH="arm64"
+        TARGET_FLOAT="hard"
+        TARGET_CPU="cortex-a55"
+        TARGET_CPU_FLAGS="+crc"
+        TARGET_FPU="crypto-neon-fp-armv8"
+        ;;
+    esac
+
+  # Kernel target
+    KERNEL_TARGET="Image"
+
+  # Graphic drivers to use
+    GRAPHIC_DRIVERS="panfrost"
+
+  # kernel serial console
+    EXTRA_CMDLINE="console=uart8250,mmio32,0xfe660000 console=tty0 coherent_pool=2M cec.debounce_ms=5000 mem=3838M"
```

Comment:
Because most/all supported boards has a proper `/chosen/stdout-path` prop, we could just add `earlycon` without any options and Linux should pick up correct configuration.

Having the `console=uart8250,mmio32,0xfe660000` ensure uart2 get full kernel startup logs when `quiet` is dropped.

> prevent double printing lines in case systemd.debug_shell is the same uart port

Do you have a simple way to reproduce this?

I have mostly seen double logs when I incorrectly had a second uart2 console added, i.e. `console=ttyS2`, something that is not needed with the first `console=uart8250` already added.

My serial debug cmdline for these boards is typically: 
```
earlycon console=uart8250,mmio32,0xfe660000 systemd.debug-shell=ttyS2 textmode ssh
```
Seem to correctly configure the early bootconsole and later hand of to a console at `ttyS2` and once kernel is running a debug shell is attached to `ttyS2`.

Reply:
> Having the console=uart8250,mmio32,0xfe660000 ensure uart2 get full kernel startup logs when quiet is dropped.

Without having this tested recently: This was always the case with only `console=mmio...` and the second `console` arg set to `console=tty0` for me, but that might have changed. I've always needed to change the second to `console=ttySX` to get a full bootlog. I really wonder if we are doing the right thing here, as having two `console` args looks just wrong. To my understanding `console=tty0`is needed for the frambuffer console used by `plymouth` - maybe it's time to switch to non-lite version of `plymouth` which supports `drm` and doesn't rely on ancient fb console :laughing: 

That being said you should probably pull in HDMI-patches for RK3588 which have been merged 6.13, as I'd expect LE to stay on expected-LTS 6.12 kernel for a while and then is PR should be GTG, as the remaining issues (iommu issue with DRAM addresses > 4 GB, no rkvdec2 driver) are not expected to be resolved soonish.